game/plugins/src/main/kotlin/gg/rsmod/plugins/content/combat/scripts/impl/ImpCombatScript.kt

```diff
@@ -28,30 +30,44 @@ object ImpCombatScript : CombatScript() {
             npc.facePawn(target)
             if (npc.moveToAttackRange(it, target, distance = 1, projectile = false) && npc.isAttackDelayReady()) {
                 if (world.random(10) > 2) {
-                    world.spawn(TileGraphic(tile = npc.tile, id = 74, height = 0))
                     it.wait(1)
+                    world.spawn(TileGraphic(tile = npc.tile, id = 74, height = 0))
                     var randomTile = getRandomTile(npc.tile)
                     while (world.collision.isClipped(randomTile)) {
                         randomTile = getRandomTile(npc.tile)
                     }
-//                    npc.moveTo(randomTile)
-                    println("Moved to x: ${randomTile.x} z: ${randomTile.z}")
-                } else {
-                    npc.prepareAttack(CombatClass.MELEE, StyleType.STAB, WeaponStyle.NONE)
-                    npc.animate(npc.combatDef.attackAnimation)
-                    npc.dealHit(target = target, formula = MeleeCombatFormula, delay = 1, type = HitType.MELEE)
+                    if (Random.nextBoolean() && npc.lock.canMove()) {
+                        npc.invisible = true
+                        npc.walkMask = npc.def.walkMask
+                        npc.walkTo(randomTile, MovementQueue.StepType.FORCED_RUN) // Teleport NPC
+                        it.wait(2)
+                        npc.invisible = false
+                        world.spawn(TileGraphic(tile = npc.tile, id = 74, height = 0))
+                        println("Moved to x: ${randomTile.x} z: ${randomTile.z}")
+                    } else {
+                        npc.prepareAttack(CombatClass.MELEE, StyleType.STAB, WeaponStyle.NONE)
+                        npc.animate(npc.combatDef.attackAnimation)
+                        npc.dealHit(target = target, formula = MeleeCombatFormula, delay = 1, type = HitType.MELEE)
+                    }
+                    npc.postAttackLogic(target)
                 }
-                npc.postAttackLogic(target)
+                it.wait(4)
+                target = npc.getCombatTarget() ?: break
             }
-            it.wait(4)
-            target = npc.getCombatTarget() ?: break
+
+            npc.resetFacePawn()
+            npc.removeCombatTarget()
         }
 
-        npc.resetFacePawn()
-        npc.removeCombatTarget()
     }
+}
 
-    private fun getRandomTile(tile: Tile) : Tile {
-        return tile.transform(x = tile.x + Random().nextInt(7), z = tile.z + Random().nextInt(7))
-    }
-}
\ No newline at end of file
+private fun getRandomTile(tile: Tile) : Tile {
```

Comment:
Does this take collision detection into account?

Reply:
Yes, it can only teleport to tiles that are walkable.